modules/margin-protocol/src/lib.rs

```diff
@@ -189,6 +217,68 @@ impl<T: Trait> Module<T> {
 		// TODO: implementation
 		unimplemented!()
 	}
+
+	fn _trader_margin_call(who: &T::AccountId) -> DispatchResult {
+		if !<MarginCalledTraders<T>>::contains_key(who) {
+			let threshold = TraderRiskThreshold::get();
+			let margin_level = Self::_margin_level(who, None)?;
+
+			if margin_level <= threshold.margin_call.into() {
+				<MarginCalledTraders<T>>::insert(who, ());
```

Comment:
You can use `_ensure_trader_safe` helper method to check if trader is currently safe, same to `_trader_become_safe`.